server/qrops.go

```diff
@@ -0,0 +1,291 @@
+	*bytes.Buffer
+}
+
+func (wc WriterCloserByteBuffer) Close() error {
+	return nil
+}
+
+func NewWriterCloserByteBuffer() *WriterCloserByteBuffer {
+	return &WriterCloserByteBuffer{bytes.NewBuffer([]byte{})}
+}
+
+type QRConfig struct {
+	DecodedQRURL    string
+	WithLogo        bool
+	CorrectionLevel qrcode.EncodeOption
+	KeyUID          string
+	ImageName       string
+	Size            int
+	Params          url.Values
+}
+
+func NewQRConfig(params url.Values, logger *zap.Logger) *QRConfig {
+	config := &QRConfig{}
+	config.Params = params
+	err := config.setQrURL()
+
+	if err != nil {
+		logger.Error("[qrops-error] error in setting QRURL", zap.Error(err))
+	}
+
+	config.setAllowProfileImage()
+	config.setErrorCorrectionLevel()
+	err = config.setSize()
+
+	if err != nil {
+		logger.Error("[qrops-error] could not convert string to int for size param ", zap.Error(err))
+	}
+
+	if config.WithLogo {
+		err = config.setKeyUID()
+
+		if err != nil {
+			logger.Error(err.Error())
+		}
+
+		config.setImageName()
+	}
+
+	return config
+}
+
+func (q *QRConfig) setQrURL() error {
+	qrURL, ok := q.Params["url"]
+
+	if !ok || len(qrURL) == 0 {
+		return errors.New("[qrops-error] no qr url provided")
+	}
+
+	decodedURL, err := base64.StdEncoding.DecodeString(qrURL[0])
+
+	if err != nil {
+		return err
+	}
+
+	q.DecodedQRURL = string(decodedURL)
+	return nil
+}
+
+func (q *QRConfig) setAllowProfileImage() {
+	allowProfileImage, ok := q.Params["allowProfileImage"]
+
+	if !ok || len(allowProfileImage) == 0 {
+		// we default to false when this flag was not provided
+		// so someone does not want to allowProfileImage on their QR Image
+		// fine then :)
+		q.WithLogo = false
+	}
+
+	LogoOnImage, err := strconv.ParseBool(allowProfileImage[0])
+
+	if err != nil {
+		// maybe for fun someone tries to send non-boolean values to this flag
+		// we also default to false in that case
+		q.WithLogo = false
+	}
+
+	// if we reach here its most probably true
+	q.WithLogo = LogoOnImage
+}
+
+func (q *QRConfig) setErrorCorrectionLevel() {
+	level, ok := q.Params["level"]
+	if !ok || len(level) == 0 {
+		// we default to MediumLevel of error correction when the level flag
+		// is not passed.
+		q.CorrectionLevel = qrcode.WithErrorCorrectionLevel(qrcode.ErrorCorrectionMedium)
+	}
+
+	levelInt, err := strconv.Atoi(level[0])
+	if err != nil || levelInt < 0 {
+		// if there is any issue with string to int conversion
+		// we still default to MediumLevel of error correction
+		q.CorrectionLevel = qrcode.WithErrorCorrectionLevel(qrcode.ErrorCorrectionMedium)
+	}
+
+	switch levelInt {
+	case 1:
+		q.CorrectionLevel = qrcode.WithErrorCorrectionLevel(qrcode.ErrorCorrectionLow)
+	case 2:
+		q.CorrectionLevel = qrcode.WithErrorCorrectionLevel(qrcode.ErrorCorrectionMedium)
+	case 3:
+		q.CorrectionLevel = qrcode.WithErrorCorrectionLevel(qrcode.ErrorCorrectionQuart)
+	case 4:
+		q.CorrectionLevel = qrcode.WithErrorCorrectionLevel(qrcode.ErrorCorrectionHighest)
+	}
```

Comment:
Perhaps add a default to this switch incase you get a result that is not 1 - 4

```suggestion
	switch levelInt {
	case 1:
		q.CorrectionLevel = qrcode.WithErrorCorrectionLevel(qrcode.ErrorCorrectionLow)
	case 2:
		q.CorrectionLevel = qrcode.WithErrorCorrectionLevel(qrcode.ErrorCorrectionMedium)
	case 3:
		q.CorrectionLevel = qrcode.WithErrorCorrectionLevel(qrcode.ErrorCorrectionQuart)
	case 4:
		q.CorrectionLevel = qrcode.WithErrorCorrectionLevel(qrcode.ErrorCorrectionHighest)
	default:
		q.CorrectionLevel = qrcode.WithErrorCorrectionLevel(qrcode.ErrorCorrectionMedium)
	}
```

Reply:
fixed here : https://github.com/status-im/status-go/pull/3172/commits/14e8bb82545b5208326ad97e1227174acc19d645